src/Cache.php

```diff
@@ -56,24 +55,30 @@ final class Cache implements CacheInterface
      */
     public function __construct(\Psr\SimpleCache\CacheInterface $handler, $defaultTtl = null)
     {
-        $this->handler = $handler;
+        $this->psr = new PsrSimpleCache($this, $handler);
```

Comment:
I think that should not be done like this. It should be a separate `DependencyAwareCache`.